appinventor/components/src/com/google/appinventor/components/runtime/PhoneStatus.java

```diff
@@ -116,6 +118,34 @@ public static boolean isConnected() {
     "process if we are using WebRTC. This is a bit of a kludge...")
   public String setHmacSeedReturnCode(String seed, String rendezvousServer) {
 
+    /*
+     * Check to see if we are being re-entered.
+     *
+     * The Companion's design is to setup communications with
+     * the user's browser and get to work. Once this process starts,
+     * enough things are in motion that it is best to quit the
+     * Companion and start a fresh copy if a different code is needed.
+     *
+     * If the same code is entered more then once, we just ignore the
+     * second attempt. This often happens when someone scans a QR Code
+     * and then presses the "Connect" Button because they do not know
+     * that they don't have to do that. Effectively we are "de-bouncing"
+     * the button.
+     *
+     */
+    if (firstSeed != null) {    // Hmm. We've been here before!
+      if (!firstSeed.equals(seed)) {
+        // Attempting to use a different seed (code)
+        // Provide a warning dialog box
+        Notifier.oneButtonAlert(Form.getActiveForm(),
```

Comment:
Is this true? It seems like bad UX to force the user to quit the app and restart it, rather than just making it a bit smarter in handling multiple codes.

Reply:
I guess in particular, if they try the WebRTC mode and then it fails and they check Legacy and generate a new code, I don't think it's a good practice to ask them to quit the app and start over. Do most of our users know how to do this?